projects/igniteui-angular/src/lib/grid/grid.component.ts

```diff
@@ -354,6 +363,34 @@ export class IgxGridComponent implements OnInit, OnDestroy, AfterContentInit, Af
             // setter called before grid is registered in grid API service
             this.sortingExpressions.unshift.apply(this.sortingExpressions, this._groupingExpressions);
         }
+        if (JSON.stringify(oldExpressions) !== JSON.stringify(newExpressions)) {
```

Comment:
Even if this is check is needed, it should be moved up so that the actual grouping is not performed while the event is being prevented.

Reply:
This check should also include the check for the existence of `this.columnList`. I still am not sure why this check is needed though. What is the case?

Reply:
@kdinev  Without this check, passing the same expressions multiple times in a row, through the API, would raise onGroupingDone every time. This is issue #2644. I will change it so that it includes the other check as well.